src/main/java/org/prebid/server/bidder/ix/IxBidder.java

```diff
@@ -290,9 +292,28 @@ private static Response mergeNativeImpTrackers(Response response, List<EventTrac
                 .build();
     }
 
-    private static BidType getBidType(String impId, List<Imp> imps) {
+    private static BidType getBidType(Bid bid, List<Imp> imps) {
+        Integer mType = bid.getMtype();
+        BidType bidType = mType != null ? switch (bid.getMtype()) {
+            case 1 -> BidType.banner;
+            case 2 -> BidType.video;
+            case 3 -> BidType.audio;
+            case 4 -> BidType.xNative;
+            default -> null;
+        } : null;
+        if (bidType != null) {
+            return bidType;
+        }
+
+        Optional<String> prebidType = Optional.ofNullable(bid.getExt())
```

Comment:
final

Reply:
If you will move logic from 315 line to separate method, then you code will be easily simplified to
```java
return Optional.ofNullable(bid.getExt())
...
.orElseGet(()->methodWithLogicFrom315Line());
```